lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/BuiltinsProcessor.java

```diff
@@ -52,36 +62,143 @@ public final boolean process(Set<? extends TypeElement> annotations, RoundEnviro
     return true;
   }
 
+  public void handleClassElement(Element element, RoundEnvironment roundEnv) throws IOException {
+    TypeElement elt = (TypeElement) element;
+    Builtin annotation = element.getAnnotation(Builtin.class);
+    String clazzName = element.getSimpleName().toString();
+    String builtinPkg =
+            annotation.pkg().isEmpty() ? BuiltinsPkg : BuiltinsPkg + "." + annotation.pkg();
+    ClassName builtinType =  new ClassName(builtinPkg, clazzName);
+    JavaFileObject gen =
+            processingEnv.getFiler().createSourceFile(builtinType.fullyQualifiedName());
+    Optional<String> stdLibName = annotation.name().isEmpty() ? Optional.empty() : Optional.of(annotation.name());
+    generateBuiltinType(gen, builtinType, stdLibName);
+  }
+
+  private void generateBuiltinType(
```

Comment:
It might be interesting to see the generated output for a particular usage. It is hard to envision that without actually trying to compile the PR.

Reply:
You mean in the comments? Because we don't show anywhere that for any processor. The generated classes are essentially your source of truth

Reply:
I added a lot of documentation as part of the rewrite I did  to the DSL. Hopefully that improves the situation.